From 99b7fbb7db1cfb2826ddcb707dfba54bd76b3b3e Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Tue, 9 Nov 2021 11:09:36 +0100
Subject: [PATCH 148/179] ARM: dovetail: add syscall_get_arg0() helper

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 arch/arm/include/asm/syscall.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm/include/asm/syscall.h b/arch/arm/include/asm/syscall.h
index fd02761ba..eec5bca5c 100644
--- a/arch/arm/include/asm/syscall.h
+++ b/arch/arm/include/asm/syscall.h
@@ -63,6 +63,11 @@ static inline void syscall_get_arguments(struct task_struct *task,
 	memcpy(args, &regs->ARM_r0 + 1, 5 * sizeof(args[0]));
 }
 
+static inline unsigned long syscall_get_arg0(struct pt_regs *regs)
+{
+	return regs->ARM_ORIG_r0;
+}
+
 static inline void syscall_set_arguments(struct task_struct *task,
 					 struct pt_regs *regs,
 					 const unsigned long *args)
-- 
2.38.1

