From f589c383c2fadee696e4fc7acb95c29d15f974bf Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Thu, 7 Apr 2022 10:28:47 +0200
Subject: [PATCH 159/179] dmaengine: imx-sdma: dovetail: add missing spinlock
 init

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 drivers/dma/imx-sdma.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/dma/imx-sdma.c b/drivers/dma/imx-sdma.c
index 43572c4e8..3648d3c62 100644
--- a/drivers/dma/imx-sdma.c
+++ b/drivers/dma/imx-sdma.c
@@ -2030,6 +2030,9 @@ static int sdma_init(struct sdma_engine *sdma)
 	clk_disable(sdma->clk_ipg);
 	clk_disable(sdma->clk_ahb);
 
+#ifdef CONFIG_IMX_SDMA_OOB
+	raw_spin_lock_init(&sdma->oob_lock);
+#endif
 	return 0;
 
 err_dma_alloc:
-- 
2.38.1

